code/datums/components/shell.dm

```diff
@@ -105,6 +105,10 @@
 		source.balloon_alert(attacker, "[locked? "locked" : "unlocked"] [source]")
 		return COMPONENT_NO_AFTERATTACK
 
+	if(attached_circuit && istype(item, /obj/item/circuit_component))
```

Comment:
Could you replace the istype check with a flag or trait?

Reply:
it'd be an istype check with extra steps in that case.

Reply:
Correct, that's what generalizing behavior is.
Instead of being a property of a single type it becomes a trait of a broader class.

In case it wasn't clear, I meant adding this on whatever type `attached_circuit` may assume.
You'd be adding the potential for those things to act as circuits, or something like that.

Reply:
I don't think I understand still.

Reply:
My bad, I misread the code. All clear!